backend/packages/Upgrade/src/api/repositories/AnalyticsRepository.ts

```diff
@@ -418,100 +411,110 @@ export class AnalyticsRepository {
-      )
-      .leftJoin(
-        StateTimeLog,
-        'enrollmentCompleteStateTimeLog',
-        `"enrollmentCompleteStateTimeLog"."experimentId" = "experiment"."id" AND enrollmentCompleteStateTimeLog.toState = 'enrollmentComplete'`
-      )
-      .leftJoin(
-        IndividualEnrollment,
-        'individualEnrollment',
-        'individualEnrollment.userId = monitoredDecisionPoint.userId'
-      )
-      .leftJoin(
-        IndividualExclusion,
-        'individualExclusion',
-        'individualExclusion.userId = monitoredDecisionPoint.userId'
-      )
-      .leftJoin(ExperimentCondition, 'condition', '"condition"."conditionCode" = monitoredDecisionPoint.condition')
-      .leftJoin(ConditionPayload, 'conditionPayload', 'conditionPayload.parentConditionId = condition.id')
-      .leftJoin(
-        DecisionPoint,
-        'decisionPointData',
-        'decisionPointData.site = monitoredDecisionPoint.site AND decisionPointData.target = monitoredDecisionPoint.target'
-      )
       .leftJoin('monitoredDecisionPoint.monitoredPointLogs', 'monitoredPointLogs')
-      .groupBy('experiment.id')
-      .addGroupBy('experiment.name')
-      .addGroupBy('"experimentCondition"."conditionCode"')
-      .addGroupBy('"monitoredDecisionPoint"."site"')
-      .addGroupBy('"monitoredDecisionPoint"."target"')
-      .addGroupBy('"monitoredDecisionPoint"."userId"')
-      .addGroupBy('"decisionPointData"."id"')
-      .addGroupBy('"individualEnrollment"."groupId"')
-      .addGroupBy('"individualExclusion"."groupId"')
-      .addGroupBy('condition."conditionCode"')
-      .addGroupBy('"decisionPointData"."excludeIfReached"')
-      .addGroupBy('"conditionPayload"."payloadValue"')
-      .addGroupBy('"enrollingStateTimeLog"."timeLog"')
-      .addGroupBy('"enrollmentCompleteStateTimeLog"."timeLog"')
-      .addGroupBy('"monitoredPointLogs"."createdAt"')
-      .addGroupBy('"individualEnrollment"."enrollmentCode"')
-      .addGroupBy('"individualExclusion"."exclusionCode"')
-      .addGroupBy('"userStratificationFactor"."stratificationFactorValue"')
-      .addGroupBy('"monitoredDecisionPoint"."userId"')
-      .orderBy('"monitoredDecisionPoint"."userId"', 'ASC')
+      .orderBy('monitoredDecisionPoint.userId', 'ASC')
       .offset(skip)
       .limit(take)
-      .where('"monitoredDecisionPoint"."experimentId" = :experimentId', { experimentId })
-      .execute();
+      .where('monitoredDecisionPoint.experimentId = :experimentId', { experimentId });
+    
+    const [individualEnrollmentQueryResults, monitoredDecisionPointQueryResults] = await Promise.all([
+      individualEnrollmentQuery.getRawMany(),
+      monitoredDecisionPointQuery.getRawMany(),
+    ]);
+
+    const userStratificationFactorUserList = [];
+
+    const individualEnrollmentExperimentData = [];
+    individualEnrollmentQueryResults.forEach(individualEnrollmentQueryResult => {
+      experimentsData.forEach((experiment) => {
```

Comment:
Why this `experimentData` is an array? This should be a single experiment

Reply:
I am preparing an array of experimentData to have information about each condition and decisionpoint, so that I can get the conditionName and conditionPayload from this array. Because from the individualEnrollment table query we just get the conditionId assigned for the user, again we either need to perform a join with this table or have a separate api call to get the condition and payload data. We get the site and target information from the monitoredDP query. Below is the array for 1 DP and 1 conditions to get an idea:

experimentsData:  [
  {
    experimentId: 'be3ae74f-370a-4015-93f3-7761d16f8b17',
    experimentName: 'Test Experiment',
    context: [ 'home' ],
    assignmentUnit: 'group',
    group: 'teacher',
    consistencyRule: 'group',
    designType: 'Simple',
    algorithmType: 'random',
    stratification: null,
    postRule: 'continue',
    revertTo: null,
    enrollmentStartDate: 2024-06-03T12:05:27.132Z,
    enrollmentCompleteDate: null,
    payload:  'payload1',
    excludeIfReached: true,
    expDecisionPointId: '2dfa7f7f-47c6-4b81-bf49-d0f41e718687',
    expConditionId: 'c22467b1-f0e9-4444-9517-cc03037bc079',
    conditionName: 'ConditionA'
  },
  {
    experimentId: 'be3ae74f-370a-4015-93f3-7761d16f8b17',
    experimentName: 'Test Experiment',
    context: [ 'home' ],
    assignmentUnit: 'group',
    group: 'teacher',
    consistencyRule: 'group',
    designType: 'Simple',
    algorithmType: 'random',
    stratification: null,
    postRule: 'continue',
    revertTo: null,
    enrollmentStartDate: 2024-06-03T12:05:27.132Z,
    enrollmentCompleteDate: null,
    payload: 'payload2',
    excludeIfReached: true,
    expDecisionPointId: '42ffaf10-6da1-4dc4-a6fe-6b54bb8cc95e',
    expConditionId: 'd22467b1-f0e9-4444-9517-cc03037bc079',
    conditionName: 'ConditionB'
  }
  ]

Reply:
Please include the types.